server/etcdserver/server.go

```diff
@@ -1051,6 +1051,19 @@ func (s *EtcdServer) run() {
 					}
 					lid := lease.ID
 					s.GoAttach(func() {
+						// If the cpu and disk io latency is high, the old leader may still revoke the lease after becoming the follower.
+						// the root cause is that etcdserver processing log entries will be blocked due to high cpu and disk io latency, which makes it impossible to clear the lease and update the membership in time.
+						// this PR solves this issue by get the leader identity directly from the raft state machine.
+						// more detail info please see issue https://github.com/etcd-io/etcd/issues/12528
+						if !s.isLeader() || !s.isLeaderInRaftStateMachine() {
```

Comment:
Can we explain why `s.isLeader()` not enough?

Reply:
updated.